### ISSUE TITLE
Outlook: body.getAsync fails with non-char ascii codes

### ISSUE THREAD

**Issue by abrahamdio** [OfficeDev/office-js#1264]
`Office.context.mailbox.item.body.getAsync` throws `SyntaxError: JSON Parse error: Unterminated string` when email content has non-char ascii codes (i.e. `&#17;`). It is observed in Outlook for Mac, and Outlook for Windows, but not in Outlook for Web (OWA). 

## Expected Behavior
`body.getAsync` should return email's content (like OWA)

## Current Behavior
- Error is thrown for Outlook for Mac
- No error thrown, and returns undefined for Outlook for Windows. 
The issue happens in read mode only (inbound email with non-char ascii codes).

## Steps to Reproduce, or Live Example
1. Open new compose
2. Use `body.setBodyAsync` to set non-char ascii characters
```
Office.context.mailbox.item.body.setAsync(
    '<span>&#17;</span><span>&#18;</span><span>&#19;</span><span>&#20;</span>',
    {coercionType: Office.CoercionType.Html},
    () => {console.log('success')});
```
3. [Unrelated] notice the `?` chars that get inserted to the compose window.
![image](https://user-images.githubusercontent.com/7740509/86654131-93936280-bf9a-11ea-85d0-0340a862ef8c.png)
4. Send the email, open the email from sent folder
5. Use `body.getAsync` to retrieve email's content in text mode
```
Office.context.mailbox.item.body.getAsync('text', (res) => {console.log(res)});
```
6. Notice the following errors.

Outlook for Mac
![image](https://user-images.githubusercontent.com/7740509/86654630-f08f1880-bf9a-11ea-8249-44d7772749ab.png)

Outlook for Windows
![image](https://user-images.githubusercontent.com/7740509/86654699-00a6f800-bf9b-11ea-91e8-e8522210a0c8.png)

## Context
There are some automated antivirus or email filtering software that appends links at the bottom of the email which sometimes contain these non-char ascii codes. In such `<a>` tag, only the link's text content (the text that gets rendered in the email) breaks the API, but not the URL on the href attribute. Currently, our add-in is failing to grab the email content.

For example:

```<a href="https://example.com/test?param&#17;">https://example.com/test?param&#17;</a>```

## Your Environment
* Platform [PC desktop, Mac, iOS, Office Online]: PC, Mac
  - Outlook for Mac (16.38), MacOS 10.15.5
  - Outlook for Windows (Version 1912, 12307.20000 C2R)
  - Both on O365
* Host: Outlook
* Office version number: Mac (16.38), Windows (Version 1912, 12308.20000 C2R)
* Operating System: MacOS 10.15.5, Windows 10 Pro (1903)
* Browser (if using Office Online): Tested the issue is **not** reproducible in OWA with Chrome browser

## Useful logs
Screenshots are attached in the repro steps

**Comment by lindalu-MSFT** [comment]
Hi @abrahamdio This programming question appears in a couple places on Stack Overflow. Please consider posting your question there to the Outlook developer community! That way, more eyes will be on your issue to help you. Let me know if you post there and please share the link so we can track it! Meanwhile, I assigned this to our Outlook team to triage. Thanks!

Stack Overflow tagged[ [office-js] and [outlook]](https://stackoverflow.com/questions/tagged/office-js+outlook)
also, questions posted about "body.getasync error" are [here](https://stackoverflow.com/search?q=body.getasync+error)

**Comment by lindalu-MSFT** [comment]
P.S. I'm sure you've already checked this topic, but posting here anyone in case it helps others!
[Office.Body interface on Office DevDocs](https://docs.microsoft.com/en-us/javascript/api/outlook/office.body?view=outlook-js-preview).

**Comment by abrahamdio** [comment]
@lindalu-MSFT thanks for responding! I am afraid this might be a regression rather than a programming question. Can you re-consider it to be triaged to the OfficeJS API team? 

As mentioned before, the use case is there are certain characters that's breaking the API contract, and we don't have control on those inserted characters because it is allowed to be inserted in the composer in the first place.

**Comment by exextoc** [comment]
Thanks for bringing this to our attention. This is a known issue in our backlog. Unfortunately, we don't have any timelines to share regarding the fix.

**Comment by JuaneloJuanelo** [comment]
We are reopning this issue is not fixed for getBody.

**Comment by Byron-Mo** [comment]
Thanks for reopening the Github issue. Here are more details of other users experiencing the problem:

Outlook version:
![image](https://user-images.githubusercontent.com/14811552/122449396-a8be9280-cf5a-11eb-915d-56525de429d1.png)

![image](https://user-images.githubusercontent.com/14811552/122449332-993f4980-cf5a-11eb-9b15-29c9b126569d.png)
![image](https://user-images.githubusercontent.com/14811552/122449363-a1978480-cf5a-11eb-843f-7b00f9942b28.png)

Please let us know if you need additional information. happy to help!

**Comment by exextoc** [comment]
Tentatively this will be fixed in Windows Builds 16.0.14225.XXXXXX and later. (this may not be an exact published build number, but builds AFTER the above should have the fix).

We will reach other to the Mac platform team, about this issue as well.

**Comment by exextoc** [comment]
Are you facing the issue for Outlook Mac Old Ui or Outlook Mac New UI ?
It works fine for Outlook Mac New UI. 
Please switch to Outlook Mac New UI.

**Comment by Byron-Mo** [comment]
> Are you facing the issue for Outlook Mac Old Ui or Outlook Mac New UI ?
> It works fine for Outlook Mac New UI.
> Please switch to Outlook Mac New UI.

That is good to know, unfortunately the impacted user is using Outlook Desktop on Windows

**Comment by exextoc** [comment]
Thanks for letting us know. 
As suggested above, tentatively, this bug will be fixed in Windows Builds 16.0.14225.XXXXXX and later.
